apps/newsletters-api/src/app/routes/newsletters.ts

```diff
@@ -79,99 +84,121 @@ export function registerReadNewsletterRoutes(app: FastifyInstance) {
 }
 
 export function registerReadWriteNewsletterRoutes(app: FastifyInstance) {
-	app.patch<{
-		Params: { newsletterId: string };
-		Body: unknown;
-	}>('/api/newsletters/:newsletterId', async (req, res) => {
-		const user = getUserProfile(req);
-		const accessDeniedError = await makeAccessDeniedApiResponse(
+	const hasAccessHook = async (
+		request: FastifyRequest,
+		reply: FastifyReply,
+	) => {
+		const user = getUserProfile(request);
+		const isAuthorised = await hasEditAccess(user.profile);
+		const isAuthorisedForUpdate = await isAuthorisedToUpdateNewsletter(
 			user.profile,
-			'editNewsletters',
+			request,
 		);
-		if (accessDeniedError) {
-			return res.status(403).send(accessDeniedError);
+		if (!isAuthorised || !isAuthorisedForUpdate) {
+			void reply.status(403).send(makeErrorResponse('You do not have edit access'));
 		}
+	};
 
-		const { newsletterId } = req.params;
-		const { body: modifications } = req;
-		const newsletterIdAsNumber = Number(newsletterId);
+	app.patch<{
+		Params: { newsletterId: string };
+		Body: unknown;
+	}>(
+		'/api/newsletters/:newsletterId',
+		{ preValidation: hasAccessHook },
```

Comment:
I've not seen this feature before! Not necessarily one for this PR, but it looks like we could use another hook to do the 'Non numeric id provided' check too

Reply:
Yeah - found it when looking at middleware for Fastify - basically it uses hooks instead and there are a bunch of events you can hook in to.